### ISSUE TITLE
Not able to create new folder in blob container (SAS, ADLS Gen2)

### ISSUE THREAD

**Issue by deepesh-pareek-by** [microsoft/AzureStorageExplorer#4040]
Hi team,

I am using SAS URI to connect to Azure account. The connection is successful but when I am trying to create a new folder in blob container it is failing again and again. My colleague is using the same URI but he is able to create but for few of us we are getting request failed. I am sharing the screenshots for the issue with you. Kindly help as this is hampering our production release performance. 
<img width="391" alt="Error" src="https://user-images.githubusercontent.com/75685695/106435988-922b0980-6499-11eb-93c5-548c2fde6281.PNG">
<img width="465" alt="SAS_URI_Connect" src="https://user-images.githubusercontent.com/75685695/106436007-97885400-6499-11eb-94e9-ed0ea823fd08.PNG">
<img width="692" alt="FolderCreationFailed" src="https://user-images.githubusercontent.com/75685695/106437869-bd165d00-649b-11eb-99bc-738752141242.PNG">

**Comment by deepesh-pareek-by** [comment]
Can anyone help on this one on urgent basis ?

**Comment by MRayermannMSFT** [comment]
@deepesh-pareek-by can you clarify if you are attaching the account or one specific container? Can you share the SAS token with us? **Please redact the SIG portion of the SAS.**

**Comment by deepesh-pareek-by** [comment]
Hi @Matthew Rayermann<mailto:notifications@github.com> / team,

We are attaching one specific container. I am sharing the SAS URL below:

https://lanapandev01.blob.core.windows.net?sv=2019-12-12&ss=bfqt&srt=sco&sp=rwlacupx&se=2031-01-16T01:47:56Z&st=2021-01-15T17:47:56Z&spr=https&sig=XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX

Moreover, I would like to bring it to your notice that few of my team mates using old verions 1v.15.1 are able to create folder but we are using 1v.17.1 and facing issue. Kindly assist.

Warm Regards,
Deepesh Pareek

**Comment by deepesh-pareek-by** [comment]
HI @MRayermannMSFT ,

Why I was able to do it in an older version (1.15.1) but not in the latest version 1.17.1 kindly assist.

**Comment by MRayermannMSFT** [comment]
@deepesh-pareek-by @JasonYeMsft is going to look into this. But, can you clarify what you mean by 1.17.1? We only released a version 1.17.0....

**Comment by JasonYeMSFT** [comment]
@deepesh-pareek-by The regression was due to us adding support to detect HNS info for a SAS attached storage account. With HNS enabled, we use data lake's create directory API to create directory so that we can create empty directories. However, the old SDK we ship with 1.17.0 doesn't support authenticating with SAS for data lake operations. We have migrated a new version of SDK and will support this scenario in our 1.18.0 release.
